### PR TITLE
fix(config): Adjust MSRV resolve config field name / values

### LINKED ISSUE #13540: Config support for controlling MSRV-aware resolver
Split out of #9930 for [RFC #3537](https://rust-lang.github.io/rfcs/3537-msrv-resolver.html) for a more focused conversation

We need a config field but this was left undefined in the RFC

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

Fixes #13540

### CODE REVIEW

File: src/doc/src/reference/unstable.md [hunk 1]
```diff
@@ -338,26 +338,26 @@ This was stabilized in 1.79 in [#13608](https://github.com/rust-lang/cargo/pull/
...
 
 `-Zmsrv-policy` allows access to an MSRV-aware resolver which can be enabled with:
-- `resolver.something-like-precedence` config field
+- `resolver.incompatible-rust-versions` config field
 - `workspace.resolver = "3"` / `package.resolver = "3"`
 - `package.edition = "2024"` (only in workspace root)
 
 The resolver will prefer dependencies with a `package.rust-version` that is the same or older than your project's MSRV.
 Your project's MSRV is determined by taking the lowest `package.rust-version` set among your workspace members.
 If there is none set, your toolchain version will be used with the intent to pick up the version from rustup's `rust-toolchain.toml`, if present.
```
Comment (weihanglo) [inline src/doc/src/reference/unstable.md:347]: Slightly off-topic, but we may not want to expose the implementation detail of rustup.

```suggestion
If there is none set, your toolchain version will be used. The toolchain version is usually determined by `rustup` or the system package manager. 
```

File: src/doc/src/reference/unstable.md [hunk 2]
```diff
@@ -338,26 +338,26 @@ This was stabilized in 1.79 in [#13608](https://github.com/rust-lang/cargo/pull/
...
 * Type: string
-* Default: "something-like-maximum"
-* Environment: `CARGO_RESOLVER_SOMETHING_LIKE_PRECEDENCE`
+* Default: "allow"
+* Environment: `CARGO_RESOLVER_INCOMPATIBLE_RUST_VERSIONS`
 
-Select which policy should be used when resolving dependencies.  Values include
-- `something-like-maximum`: prefer highest compatible versions of a package
-- `something-like-rust-version`: prefer versions of packages compatible with your project's Rust version
+Select how packages with incompatible rust-versions should be resolved.  Values include:
```
Comment (weihanglo) [inline src/doc/src/reference/unstable.md:354]: It is unclear what an incompatible rust-version is. Should we link "incompatible rust-versions" to `package.rust-version` in the manifest?

File: src/doc/src/reference/unstable.md [hunk 3]
```diff
@@ -338,26 +338,26 @@ This was stabilized in 1.79 in [#13608](https://github.com/rust-lang/cargo/pull/
...
+- `allow`: treat them like any other dependency
```
Comment (weihanglo) [inline src/doc/src/reference/unstable.md:355]: This is also unclear as user doesn't know how other dependency is treated. Are we making it vague for rooms for future behavior changes?

File: src/doc/src/reference/unstable.md [hunk 4]
```diff
@@ -338,26 +338,26 @@ This was stabilized in 1.79 in [#13608](https://github.com/rust-lang/cargo/pull/
...
+- `fallback`: only consider them if no other dependency version matched
 
 Can be overridden with
 - `--ignore-rust-version` CLI option
-- Setting the dependency's version requirement too high
+- Setting the dependency's version requirement above any version that is rust-version compatible
```
Comment (weihanglo) [inline src/doc/src/reference/unstable.md:360]: Should we highlight the field name?

```suggestion
- Setting the dependency's version requirement above any version with a compatible `package.rust-version`
```

File: src/doc/src/reference/unstable.md (hunk 1 above)
Comment (epage) [reply src/doc/src/reference/unstable.md:347]: This section isn't as important long-term as this is just describing the unstable feature.

However, in describing it this way, I'm trying to accomplish two things
- Emphasize the motivating use case for this functionality
- Help rustup users discover this by using terms they would expect

I am making a slight tweak to it so hopefully that will help.

File: src/doc/src/reference/unstable.md (hunk 3 above)
Comment (epage) [reply src/doc/src/reference/unstable.md:355]: I can't think of a reason to be intentionally vague about `allow`, I'm just unsure of what more needs to be described.

File: src/doc/src/reference/unstable.md (hunk 2 above)
Comment (epage) [reply src/doc/src/reference/unstable.md]: I've called out the field name.  We should eventually turn all of those into links.

**Comment by epage** [comment]
Removing the draft status as I've given time for some community input with only a passing reference to not liking something but the person didn't engage to get more details.

**Comment by weihanglo** [comment]
> Removing the draft status as I've given time for some community input with only a passing reference to not liking something but the person didn't engage to get more details.

Where can I find this?

File: src/doc/src/reference/unstable.md (hunk 3 above)
Comment (weihanglo) [reply src/doc/src/reference/unstable.md:356]: I think the problem is pronouns and “other”. What is included in “other”? Maybe something like


```suggestion
- `allow`: treat `package.rust-version` incompatible dependency as compatible
- `fallback`: only consider `package.rust-version` incompatible dependency if no other compatible dependency version matched
```

Pronouns are often confusing, and I prefer to use them less.

(Maybe it is just me 😬)

**Comment by epage** [comment]
> > Removing the draft status as I've given time for some community input with only a passing reference to not liking something but the person didn't engage to get more details.
> 
> Where can I find this?

From https://hachyderm.io/@djc/112842279033657989

> [@epage](https://hachyderm.io/@epage) so the current behavior is allow and the new default would be fallback, right? I don’t love fallback because it describes a sort of meta-behavior that the resolver implements rather than a decision on the particular package version. From that perspective, I feel like “ignore” might be a better fit? “When Cargo encounters a dependency version with an incompatible rust-version, it allows/ignores that version.”

File: src/doc/src/reference/unstable.md (hunk 3 above)
Comment (epage) [reply src/doc/src/reference/unstable.md]: To be fully correct, its an "incompatible dependency version" (not just dependency).  This starts to become a mouthful.  I gave it a try by skipping on `package.` and just saying "version".

**Review by weihanglo — APPROVED** [review]
This seems fine to merge now. We will do an FCP when stabilizing it.

**Pull request merged**